app/BUILD.bazel

```diff
@@ -510,125 +523,137 @@ kt_android_library(
-        "src/sharedTest/java/org/oppia/app/recyclerview/RecyclerViewMatcher.kt",
-        "src/sharedTest/java/org/oppia/app/utility/ProgressMatcher.kt",
         "src/sharedTest/java/org/oppia/app/parser/RichTextViewMatcher.kt",
+        "src/sharedTest/java/org/oppia/app/recyclerview/RecyclerViewMatcher.kt",
+        "src/sharedTest/java/org/oppia/app/utility/ClickActions.kt",
+        "src/sharedTest/java/org/oppia/app/utility/DragViewAction.kt",
         "src/sharedTest/java/org/oppia/app/utility/DrawableMatcher.kt",
         "src/sharedTest/java/org/oppia/app/utility/EspressoTestsMatchers.kt",
-        "src/sharedTest/java/org/oppia/app/utility/TabMatcher.kt",
         "src/sharedTest/java/org/oppia/app/utility/FontSizeMatcher.kt",
         "src/sharedTest/java/org/oppia/app/utility/MockitoKotlinHelper.kt",
-        "src/sharedTest/java/org/oppia/app/utility/DragViewAction.kt",
-        "src/sharedTest/java/org/oppia/app/utility/ClickActions.kt",
+        "src/sharedTest/java/org/oppia/app/utility/OrientationChangeAction.kt",
+        "src/sharedTest/java/org/oppia/app/utility/ProgressMatcher.kt",
+        "src/sharedTest/java/org/oppia/app/utility/TabMatcher.kt",
     ],
     deps = [
         ":app",
         "//testing",
         artifact("org.mockito:mockito-core"),
-    ]
+    ],
+    testonly = True,
 )
 
 TEST_DEPS = [
     ":app",
     ":dagger",
-    "//utility",
-    "//testing",
-    "//domain",
     ":resources",
     ":test_deps",
+    "//domain",
+    "//testing",
+    "//utility",
     "@robolectric//bazel:android-all",
-    artifact("org.jetbrains.kotlin:kotlin-reflect"),
-    artifact("com.google.truth:truth"),
-    artifact("androidx.test.ext:junit"),
-    artifact("androidx.databinding:databinding-runtime"),
+    artifact("androidx.annotation:annotation"),
     artifact("androidx.databinding:databinding-adapters"),
     artifact("androidx.databinding:databinding-common"),
-    artifact("androidx.annotation:annotation"),
     artifact("androidx.databinding:databinding-compiler"),
-    artifact("androidx.test:runner:1.2.0"),
-    artifact("org.jetbrains.kotlinx:kotlinx-coroutines-test:1.2.2"),
-    artifact("org.robolectric:annotations:4.3"),
-    artifact("org.mockito:mockito-core"),
+    artifact("androidx.databinding:databinding-runtime"),
     artifact("androidx.test.espresso:espresso-contrib:3.1.0"),
     artifact("androidx.test.espresso:espresso-core:3.2.0"),
     artifact("androidx.test.espresso:espresso-intents:3.1.0"),
+    artifact("androidx.test.ext:junit"),
+    artifact("androidx.test:runner:1.2.0"),
+    artifact("com.google.truth:truth"),
+    artifact("org.jetbrains.kotlin:kotlin-reflect"),
+    artifact("org.jetbrains.kotlinx:kotlinx-coroutines-test:1.2.2"),
+    artifact("org.mockito:mockito-core"),
+    artifact("org.robolectric:annotations:4.3"),
 ]
 
+'''
+Tests to be processed using the test genrule. Test files should be processed if they import resource
```

Comment:
Should this say that all test files should go here, instead?

Reply:
Not all test files are using the genrule, though. I am updating this documentation anyway (to be consistent with documentation I wrote a few PRs back) so please let me know if it is clearer now.

Reply:
Much clearer--thanks!